modules/react/popup/lib/hooks/useFocusRedirect.ts

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+import {PopupModel} from '../usePopupModel';
+import {getFirstFocusableElement, getLastFocusableElement} from '@workday/canvas-kit-react/common';
+
+/**
+ * "Trap" or "loop" focus within a PopupModel's `stackRef` element. This is required for
+ * accessibility on modals. If a keyboard users hits the Tab or Shift + Tab, this will force
+ * "looping" of focus. It effectively "hides" outside content from keyboard users. Use an overlay to
+ * hide content from mouse users and `useAssistiveHideSiblings` to hide content from assistive
+ * technology users.
+ */
+export const useFocusRedirect = (model: PopupModel, elemProps = {}) => {
+  const onKeyDown = React.useCallback(
+    (event: KeyboardEvent): void => {
+      if (model.state.stackRef.current && event.key === 'Tab') {
```

Comment:
This breaks if you shift + tab while on the last focusable element in the popup

Reply:
Excellent catch! I'll fix and add Cypress specifications.